container-search/src/main/java/com/yahoo/search/yql/MinimalQueryInserter.java

```diff
@@ -35,44 +40,61 @@
 
     private static final CompoundName MAX_HITS = new CompoundName("maxHits");
     private static final CompoundName MAX_OFFSET = new CompoundName("maxOffset");
+    private static Logger log = Logger.getLogger(MinimalQueryInserter.class.getName());
 
-    @Override
-    public Result search(Query query, Execution execution) {
-        if (query.properties().get(YQL) == null) return execution.search(query);
+    @Inject
+    public MinimalQueryInserter(Linguistics linguistics) {
+        warmup(linguistics);
+    }
+    MinimalQueryInserter() {
+        this(new SimpleLinguistics());
+    }
+    static boolean warmup() {
+        return warmup(new SimpleLinguistics());
+    }
+    static boolean warmup(Linguistics linguistics) {
```

Comment:
Could you add a comment which explains why we want to do warmup here?

Reply:
Fixed